# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/DecodedStream.h
# Commit: 1c45b135318d
# Full Hash: 1c45b135318d8cb519b8d58af72ab864fe6bfd62
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-11-14 17:12:31
# Regressor Bug: 1172394
# File Overlap Count: 1
# Description:
#   Bug 1172394 - Simplify MediaSink somewhat. r=padenot
#   
#   This patches does several minor things:
#   - Moves SetSink (from setSinkid) to automatic coalescing of multiple calls
#     through a Canonical/Mirror setup instead of a manual atomic counter.
# ==============================================================================

diff -r c57c41e8c39e -r 1c45b135318d dom/media/mediasink/DecodedStream.h
--- a/dom/media/mediasink/DecodedStream.h	Wed Nov 13 08:55:39 2019 +0000
+++ b/dom/media/mediasink/DecodedStream.h	Wed Nov 13 08:55:54 2019 +0000
@@ -33,18 +33,13 @@
 class MediaQueue;
 
 class DecodedStream : public MediaSink {
-  using MediaSink::PlaybackParams;
-
  public:
   DecodedStream(MediaDecoderStateMachine* aStateMachine,
                 nsTArray<RefPtr<ProcessedMediaTrack>> aOutputTracks,
+                double aVolume, double aPlaybackRate, bool aPreservesPitch,
                 MediaQueue<AudioData>& aAudioQueue,
                 MediaQueue<VideoData>& aVideoQueue);
 
-  // MediaSink functions.
-  const PlaybackParams& GetPlaybackParams() const override;
-  void SetPlaybackParams(const PlaybackParams& aParams) override;
-
   RefPtr<EndedPromise> OnEnded(TrackType aType) override;
   media::TimeUnit GetEndTime(TrackType aType) const override;
   media::TimeUnit GetPosition(TimeStamp* aTimeStamp = nullptr) const override;
@@ -58,6 +53,8 @@
   void SetPreservesPitch(bool aPreservesPitch) override;
   void SetPlaying(bool aPlaying) override;
 
+  double PlaybackRate() const override;
+
   nsresult Start(const media::TimeUnit& aStartTime,
                  const MediaInfo& aInfo) override;
   void Stop() override;
@@ -100,7 +97,9 @@
   Mirror<PrincipalHandle> mPrincipalHandle;
   const nsTArray<RefPtr<ProcessedMediaTrack>> mOutputTracks;
 
-  PlaybackParams mParams;
+  double mVolume;
+  double mPlaybackRate;
+  bool mPreservesPitch;
 
   media::NullableTimeUnit mStartTime;
   media::TimeUnit mLastOutputTime;